### PR TITLE
Allow lists as indata for bytes fields.

### PR DESCRIPTION
**Pull request by lukebakken**
Another piece of `epb` compatibility.

See tomas-abrahamsson/gpb#63

TODO:

- [x] NIF support

### CODE REVIEW

**Comment by lukebakken** [comment]
@tomas-abrahamsson - I added the following nif test, expecting it to fail, but it didn't:

https://github.com/basho/gpb/blob/feature/lrb/epb-encode-list-bytes-compat/test/gpb_compile_tests.erl#L1681-L1704

I thought I would have to add something here, but apparently the list must be converted to a binary before reaching this point:

https://github.com/basho/gpb/blob/feature/lrb/epb-encode-list-bytes-compat/src/gpb_compile.erl#L6670-L6678

Am I missing something?

**Comment by tomas-abrahamsson** [comment]
*blush* ahemm, mea culpa. Thanks! On the branch verify-separate-vm-catches-errors (to be merged to master soon) there's a fix plus a unit test to make sure such errors do not go unnoticed again. With that fix, your newly added eunit test fails as expected.

A useful trick for debugging such issues is to change `with_tmpdir(fun...end)` into `with_tmpdir(save, fun..end)` and you'll get a debug printout, and in the case of nif code, some C++ code, and a Makefile for further debugging. In that directory, there's also a file, _fun-run-result,_ which in this case actually did contain an (undetected due to the bug) error.

**Comment by tomas-abrahamsson** [comment]
Easiest for you might be to either rebase your branch on top of this, or cherry-pick that commit into your branch (or just temporarily-while-developing change the line to fixes the issue)

**Comment by lukebakken** [comment]
@tomas-abrahamsson thank you, I'll look further into this :smile:

**Comment by lukebakken** [comment]
@tomas-abrahamsson - all set. I rebased this branch onto `verify-separate-vm-catches-errors` and fixed the last bits of code that needed fixing. All set for final review.

File: test/gpb_compile_tests.erl [hunk 1]
```diff
@@ -1387,6 +1399,7 @@ nif_code_test_() ->
          {"Nif enums in msgs", fun nif_enum_in_msg/0},
          {"Nif enums with pkgs", fun nif_enum_with_pkgs/0},
          {"Nif with strbin", fun nif_with_strbin/0},
+         {"Nif with list indata for bytes", fun nif_with_list_indata_for_bytes/0},
```
Comment (tomas-abrahamsson) [inline test/gpb_compile_tests.erl]: This line is >80 characters

**Comment by tomas-abrahamsson** [comment]
Looks good

Small issues: A comment on length of one line, and maybe we can squash the changes into one commit? (it'll help any future git bisect if there isn't a failing eunit test in one of the commits (I know it is deliberate during development; tdd))

**Comment by tomas-abrahamsson** [comment]
Hmm... just remembered... the eunit test wasn't failing after all... :)

Another thing: I can add a sentence to the README after I've merged them all, about iolist being ok for encoding. Since the README table rewrite is in the other branch, I figure it might be easier to add that after merge.

**Comment by lukebakken** [comment]
@tomas-abrahamsson - all set I think, line length addressed, commits squashed. If you could add something to the `README` that would be great. Have a good weekend.

**Comment by tomas-abrahamsson** [comment]
Looks good, thanks for your patience, and have a nice weekend, too!

**Comment by tomas-abrahamsson** [comment]
This is now included in 3.24.0

**Pull request merged**